lightning/src/ln/msgs.rs

```diff
@@ -485,6 +496,11 @@ impl Writeable for NetAddress {
 				version.write(writer)?;
 				port.write(writer)?;
 			}
+			&NetAddress::DNSHostname { ref hostname, ref port } => {
+				5u8.write(writer)?;
+				hostname.write(writer)?;
```

Comment:
The default `String` write writes a two-byte length, whereas we need one here. You'll need to manually write the string. That said, you may actually want to use a different type, given we're required to write out hostnames in ASCII, and the tight size length, we'll want to enforce that on users.